inbm/cloudadapter-agent/cloudadapter/cloud/client/connections/mqtt_connection.py

```diff
@@ -95,8 +95,10 @@ def _set_proxy(self, config: ProxyConfig) -> None:
         @param config: (ProxyConfig) Config to use
         """
         if config.endpoint:
-            socks.set_default_proxy(socks.PROXY_TYPE_HTTP, *config.endpoint)
-            socket.socket = socks.socksocket  # type: ignore
+            proxy_addr, proxy_port = config.endpoint
+            self._client.proxy_set(proxy_type=socks.HTTP,
```

Comment:
<h3>$\color{red}Defects: ErrorHandling$</h3>

Consider adding error handling for the new proxy settings code to gracefully handle potential exceptions when setting the proxy on the MQTT client.